### PR TITLE
feat: create hooks to redux component

### PR DESCRIPTION
**Pull request by AAfghahi**
### SUMMARY
I moved the useSelector out of Database modal and created a typedSelector in state.ts. I also put RootState into types so that it is reusable. 

I feel like this is what the ticket was saying, but I am still unsure. 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/views/CRUD/data/database/state.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { useSelector, TypedUseSelectorHook } from 'react-redux';
+import { RootState } from './types';
+
+export const useTypedSelector: TypedUseSelectorHook<RootState> = useSelector;
```
Comment (eschutho) [inline superset-frontend/src/views/CRUD/data/database/state.ts:23]: essentially what you want to do here is just pull the RootState (state.common.conf) out of redux like this: https://github.com/apache/superset/blob/master/superset-frontend/src/dashboard/components/nativeFilters/state.ts#L25

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal.tsx [hunk 2]
```diff
@@ -143,7 +132,7 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
   const [db, setDB] = useState<DatabaseObject | null>(null);
   const [isHidden, setIsHidden] = useState<boolean>(true);
   const [tabKey, setTabKey] = useState<string>(DEFAULT_TAB_KEY);
-  const conf = useSelector((state: RootState) => state.common.conf);
+  const conf = useTypedSelector((state: RootState) => state.common.conf);
```
Comment (eschutho) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal.tsx:135]: here's an example of using applying the redux state to this component: https://github.com/apache/superset/blob/master/superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterBar.tsx#L185

File: superset-frontend/src/views/CRUD/data/database/state.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { useSelector } from 'react-redux';
+import { ViewState } from '../../../types';
```
Comment (eschutho) [inline superset-frontend/src/views/CRUD/data/database/state.ts:21]: if you have to go out more than one folder, you can use the alias instead, so `from 'src/views/types'`

**Review by eschutho — APPROVED** [review]
Looks great. Just one small nit, but you're good to go.

**Comment by eschutho** [comment]
@AAfghahi you can try closing and reopening this pr to see if it fixes the tests..

**Comment by AAfghahi** [comment]
Screen Shot 2021-03-12 at 10 55 45 AM
Tested this, and it is working fine. Is this the right place to look?

**Comment by AAfghahi** [comment]
> @AAfghahi you can try closing and reopening this pr to see if it fixes the tests..

That worked, thank you!

**Review by betodealmeida — APPROVED** [review]

**Pull request merged**